styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -104,31 +104,34 @@ public BackfillResource(Storage storage) {
 
   private BackfillsPayload getBackfills(RequestContext requestContext) {
     final Optional<String> componentOpt = requestContext.request().parameter("component");
+    final Optional<String> workflowOpt = requestContext.request().parameter("workflow");
     final Optional<String> statusesFlagOpt = requestContext.request().parameter("status");
 
-    final List<BackfillPayload> backfillPayloads;
-    List<Backfill> backfills;
+    Stream<Backfill> backfills;
     try {
-      backfills = storage.backfills();
+      backfills = storage.backfills().stream();
```

Comment:
Do we need to filter already in datastore instead of reading everything out? If we don't see we will have many backfills, it is OK then.